### PR TITLE
[graviton][pytorch][build]Split graviton pytorch 1.10 e3 sagemaker

### PR DESCRIPTION
**Pull request by Qingzi-Lan**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron"/"graviton">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/inference/docker/build_artifacts/torchserve-e3-entrypoint.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2019-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# Copyright 2019-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (arjkesh) [inline pytorch/inference/docker/build_artifacts/torchserve-e3-entrypoint.py:1]: nice catch, thanks

**Comment by tejaschumbalkar** [comment]
I believe the docker file name should be `Dockerfile.graviton.e3.cpu` so that we can extend the naming convention to GPU

File: pytorch/inference/docker/1.10/py3/Dockerfile.e3.graviton [hunk 2]
```diff
@@ -130,8 +130,8 @@ RUN useradd -m model-server \
  && mkdir -p /home/model-server/tmp /opt/ml/model \
  && chown -R model-server /home/model-server /opt/ml/model
 
-COPY torchserve-entrypoint.py /usr/local/bin/dockerd-entrypoint.py
-COPY config.properties /home/model-server
+COPY torchserve-e3-entrypoint.py /usr/local/bin/dockerd-entrypoint.py
+COPY config.properties /home/model-server/config.properties
```
Comment (tejaschumbalkar) [inline pytorch/inference/docker/1.10/py3/Dockerfile.e3.graviton:134]: do we need additional `/config.properties`?

File: pytorch/inference/docker/1.10/py3/Dockerfile.e3.graviton [hunk 3]
```diff
@@ -152,4 +152,4 @@ RUN curl https://aws-dlc-licenses.s3.amazonaws.com/pytorch-1.10/license.txt -o /
 
 EXPOSE 8080 8081
 ENTRYPOINT ["python", "/usr/local/bin/dockerd-entrypoint.py"]
-CMD ["torchserve", "--start", "--ts-config", "/home/model-server/config.properties", "--model-store", "/home/model-server/"]
+CMD ["torchserve", "--start", "--ts-config", "/home/model-server/config.properties"]
```
Comment (tejaschumbalkar) [inline pytorch/inference/docker/1.10/py3/Dockerfile.e3.graviton:155]: same here

File: pytorch/buildspec-graviton.yml [hunk 4]
```diff
@@ -25,14 +25,14 @@ context:
       target: config.properties
 
 images:
-  BuildGravitonPTInferencePy3DockerImage:
+  BuildE3PytorchGravitonCPUInferencePy3DockerImage:
     <<: *INFERENCE_REPOSITORY
     build: &PYTORCH_GRAVITON_INFERENCE_PY3 false
     image_size_baseline: 10000
     device_type: &DEVICE_TYPE graviton
```
Comment (tejaschumbalkar) [inline pytorch/buildspec-graviton.yml:32]: modify the device type to `cpu`

File: pytorch/inference/docker/1.10/py3/Dockerfile.e3.graviton (hunk 2 above)
Comment (Qingzi-Lan) [reply pytorch/inference/docker/1.10/py3/Dockerfile.e3.graviton:134]: I saw some other places wrote it explicitly. in case the original file name change, we can keep docker file name consistant.

File: pytorch/inference/docker/1.10/py3/Dockerfile.e3.graviton (hunk 3 above)
Comment (Qingzi-Lan) [reply pytorch/inference/docker/1.10/py3/Dockerfile.e3.graviton:155]: we can have `CMD["serve"]` here, but will need to modify the entrypoint.py to match the open source behaviour

**Comment by tejaschumbalkar** [comment]
Few test failures found related to pytest. I will cover the same in #1514

**Review by tejaschumbalkar — APPROVED** [review]
